src/testresult.h

```diff
@@ -0,0 +1,45 @@
+#ifndef TESTRESULT_H
+#define TESTRESULT_H
+
+#include <QObject>
+
+const QString REGTOKEN_KEY = QStringLiteral("regtoken");
+
+class TestResult : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY(QString testClassLabel READ testClassLabel NOTIFY testResultChanged)
+
+public:
+    enum class Result
+    {
+        NoResult,
+        Positive,
+        Negative
+    };
```

Comment:
Since this is available from QML, it would be good to add `Q_ENUM(Result)` here so the enum values are also available in the QML.
```suggestion
    };
    Q_ENUM(Result)
```

Reply:
Afaik you would still need to convert the enum to a string in QML. This is currently covered with `TestResult::testClassLabel` with already return a result QString. Does making the enum  available from QML still sense? I'm not insisting on keeping this unavailable.

Reply:
Yes, you're right, there's not much benefit given that the enum isn't really used outside the class (it could even be private).

It might be worth considering providing a `testClass` property as an enum value as well, since this is easier to reason with in the QML:
```
    Q_PROPERTY(Result testClass READ testClass NOTIFY testResultChanged)
```
Clearly this isn't needed at the moment though.